python/ray/data/tests/test_dataset.py

```diff
@@ -5532,6 +5532,11 @@ def test_nowarning_execute_with_cpu(ray_start_cluster_init):
         mock_logger.assert_not_called()
 
 
+def test_new_execution_backend_invocation(ray_start_regular_shared):
+    ds = ray.data.range(10).randomize_block_order()
+    assert ds.take_all() == list(range(10))
```

Comment:
Should this test be moved to the bulk exec tests with the flag forced on? I guess it doesn't matter if we are enabling it by default in a followup.

Reply:
Yeah, let's move to bulk exec tests.

Reply:
Sounds good, moved over there.